remote-storage-managers/hdfs/src/main/java/org/apache/kafka/rsm/hdfs/LogSegmentDataHeader.java

```
@@ -0,0 +1,205 @@
+        SEGMENT((byte) 5);
+
+        private final byte id;
+
+        FileType(byte id) {
+            this.id = id;
+        }
+
+        public static FileType fromId(byte id) {
+            switch (id) {
+                case 0:
+                    return OFFSET_INDEX;
+                case 1:
+                    return TIMESTAMP_INDEX;
+                case 2:
+                    return LEADER_EPOCH_CHECKPOINT;
+                case 3:
+                    return PRODUCER_SNAPSHOT;
+                case 4:
+                    return TRANSACTION_INDEX;
+                case 5:
+                    return SEGMENT;
+                default:
+                    return null;
+            }
+        }
+    }
+
+    public static final Integer LENGTH = 25;
+    public static final byte CURRENT_VERSION = 0;
+
+    private byte version;
+    private final EnumMap<FileType, Integer> filePositions = new EnumMap<>(FileType.class);
+
+    private LogSegmentDataHeader() {
+    }
+
+    @VisibleForTesting
+    byte version() {
+        return version;
+    }
+
+    @VisibleForTesting
+    EnumMap<FileType, Integer> filePositions() {
+        return filePositions;
+    }
+
+    public DataPosition getDataPosition(final FileType fileType) {
+        final Integer position = filePositions.get(fileType);
+        switch (fileType) {
+            case OFFSET_INDEX:
+            case TIMESTAMP_INDEX:
+            case LEADER_EPOCH_CHECKPOINT:
+            case PRODUCER_SNAPSHOT:
+            case TRANSACTION_INDEX:
+                final FileType nextFileType = FileType.fromId((byte) (fileType.id+1));
+                final int nextFilePosition = filePositions.get(nextFileType);
+                final int length = nextFilePosition - position;
+                return new DataPosition(position, length);
+            case SEGMENT:
+                return new DataPosition(position, Integer.MAX_VALUE);
+            default:
+                return null;
```

You may want to throw an `IllegalArgumentException` instead of returning null.

---

Ack.